### PR TITLE
[CORDA-2172]: Add link to error-codes-webapp in Corda.

### PR DESCRIPTION
**Pull request by sollecitom**
https://r3-cev.atlassian.net/browse/CORDA-2172

### CODE REVIEW

File: tools/cliutils/src/main/kotlin/net/corda/cliutils/CordaVersionProvider.kt [hunk 1]
```diff
@@ -9,12 +9,29 @@ import picocli.CommandLine
...
+            val raw = releaseVersion
+            val parts = raw.removeSuffix("-SNAPSHOT").split(".")
+            try {
+                "${parts[0]}.${parts[1]}${parts.getOrNull(2)?.let { ".$it" } ?: ""}"
+            } catch (e: Exception) {
+                current_major_release
+            }
+        }
+
+        internal val platformEditionCode: String by lazy { editionCodes[edition] ?: source_edition }
```
Comment (tlil) [inline tools/cliutils/src/main/kotlin/net/corda/cliutils/CordaVersionProvider.kt]: Perhaps clarify this a bit more. Currently `source_edition` defaults to a short version of the edition (i.e., `OS`). However, if I read this correctly, `manifestValue("Corda-Edition")` gets a long version (`Open Source` or `Enterprise`).

So here you end up looking up `editionCodes["OS"]` if the manifest value is non-existent. But the map maps long editions to short ones. So they're inconsistent. I know you end up with the same result. But .....

File: tools/cliutils/src/main/kotlin/net/corda/cliutils/CordaVersionProvider.kt [hunk 2]
```diff
@@ -9,12 +9,29 @@ import picocli.CommandLine
  */
 class CordaVersionProvider : CommandLine.IVersionProvider {
     companion object {
+        const val current_major_release = "4.0"
```
Comment (tlil) [inline tools/cliutils/src/main/kotlin/net/corda/cliutils/CordaVersionProvider.kt]: Why did you have to hard-code a fallback value here? Is there a missing manifest somewhere?

File: tools/cliutils/src/main/kotlin/net/corda/cliutils/ExceptionsErrorCodeFunctions.kt [hunk 3]
```diff
@@ -8,11 +8,13 @@ import java.util.*
 internal fun Message.withErrorCodeFor(error: Throwable?, level: Level): Message {
 
     return when {
-        error != null && level.isInRange(Level.FATAL, Level.WARN) -> CompositeMessage("$formattedMessage [errorCode=${error.errorCode()}]", format, parameters, throwable)
+        error != null && level.isInRange(Level.FATAL, Level.WARN) -> CompositeMessage("$formattedMessage [errorCode=${error.errorCode()}, errorDescription=${error.errorCodeLocationUrl()}]", format, parameters, throwable)
```
Comment (tlil) [inline tools/cliutils/src/main/kotlin/net/corda/cliutils/ExceptionsErrorCodeFunctions.kt]: `errorDescription` = URL? Perhaps use a different name for the key here.

File: tools/cliutils/src/main/kotlin/net/corda/cliutils/CordaVersionProvider.kt (hunk 2 above)
Comment (sollecitom) [reply tools/cliutils/src/main/kotlin/net/corda/cliutils/CordaVersionProvider.kt]: It wouldn't work from Intellij, etc.

File: tools/cliutils/src/main/kotlin/net/corda/cliutils/CordaVersionProvider.kt [hunk 4]
```diff
@@ -9,12 +9,29 @@ import picocli.CommandLine
...
         private fun manifestValue(name: String): String? = if (Manifests.exists(name)) Manifests.read(name) else null
 
-        val releaseVersion: String by lazy { manifestValue("Corda-Release-Version") ?: "Unknown" }
+        val releaseVersion: String by lazy { manifestValue("Corda-Release-Version") ?: current_major_release }
+        val edition: String by lazy { manifestValue("Corda-Edition") ?: source_edition }
         val revision: String by lazy { manifestValue("Corda-Revision") ?: "Unknown" }
         val vendor: String by lazy { manifestValue("Corda-Vendor") ?: "Unknown" }
         val platformVersion: Int by lazy { manifestValue("Corda-Platform-Version")?.toInt() ?: 1 }
+
+        internal val semanticVersion: String by lazy {
```
Comment (anthonykeenan) [inline tools/cliutils/src/main/kotlin/net/corda/cliutils/CordaVersionProvider.kt]: As discussed, I think the release version should be used in the error codes url to distinguish between released binaries and locally compiled/snapshot binaries, as we are likely to grep the website logs for frequently ocurring issues in our released versions, which we wouldn't want to get confused with development issues.

File: tools/cliutils/src/main/kotlin/net/corda/cliutils/CordaVersionProvider.kt (hunk 1 above)
Comment (sollecitom) [reply tools/cliutils/src/main/kotlin/net/corda/cliutils/CordaVersionProvider.kt]: Fixeed

File: tools/cliutils/src/main/kotlin/net/corda/cliutils/ExceptionsErrorCodeFunctions.kt (hunk 3 above)
Comment (sollecitom) [reply tools/cliutils/src/main/kotlin/net/corda/cliutils/ExceptionsErrorCodeFunctions.kt]: Renamed to `moreInformationAt`.

File: tools/cliutils/src/main/kotlin/net/corda/cliutils/CordaVersionProvider.kt (hunk 4 above)
Comment (sollecitom) [reply tools/cliutils/src/main/kotlin/net/corda/cliutils/CordaVersionProvider.kt]: Fixed

Comment (sollecitom) [reply tools/cliutils/src/main/kotlin/net/corda/cliutils/CordaVersionProvider.kt]: Also, modified the error codes web app to handle "-SNAPSHOT" suffix.

**Pull request merged**